runtime/gc_vlhgc/CopyForwardCompactGroup.hpp

```diff
@@ -132,6 +137,49 @@ class MM_CopyForwardCompactGroup
 	{
 		_TLHRemainderBase = base;
 		_TLHRemainderTop = top;
+		_TLHRemainderCount += 1;
+	}
+
+	MMINLINE uintptr_t getTLHRemainderSize()
+	{
+		return ((uintptr_t)_TLHRemainderTop - (uintptr_t)_TLHRemainderBase);
+	}
+
+	MMINLINE void discardTLHRemainder(MM_EnvironmentVLHGC* env)
+	{
+		if (NULL != _TLHRemainderBase) {
+			/* make it a walkable hole */
+			env->_cycleState->_activeSubSpace->abandonHeapChunk(_TLHRemainderBase, _TLHRemainderTop);
+			MM_GCExtensions *extensions =MM_GCExtensions::getExtensions(env);
+			MM_HeapRegionDescriptorVLHGC *region = (MM_HeapRegionDescriptorVLHGC*)extensions->heapRegionManager->tableDescriptorForAddress(_TLHRemainderBase);
+			discardHeapChunk(_TLHRemainderBase, _TLHRemainderTop, region);
+			resetTLHRemainder();
+		} else {
+			Assert_MM_true(NULL == _TLHRemainderTop);
+		}
+	}
+
+	MMINLINE void recycleTLHRemainder(MM_EnvironmentVLHGC* env)
+	{
+		if (NULL != _TLHRemainderBase) {
+			/* make it a walkable hole */
+			env->_cycleState->_activeSubSpace->abandonHeapChunk(_TLHRemainderBase, _TLHRemainderTop);
+			MM_GCExtensions *extensions =MM_GCExtensions::getExtensions(env);
+			MM_HeapRegionDescriptorVLHGC *region = (MM_HeapRegionDescriptorVLHGC*)extensions->heapRegionManager->tableDescriptorForAddress(_TLHRemainderBase);
+			MM_MemoryPoolAddressOrderedList *pool = (MM_MemoryPoolAddressOrderedList *)region->getMemoryPool();
```

Comment:
I'd be ok removing this cast. The fact that we use MPAOL should be contained within very few classes, ideally only with RegionDescriptorVLHC, although we already broke that encapsulation at a few spots in CFScheme...